src/main/java/org/opensearch/ad/rest/handler/IndexAnomalyDetectorActionHandler.java

```diff
@@ -164,22 +166,41 @@ public IndexAnomalyDetectorActionHandler(
         this.adTaskManager = adTaskManager;
     }
 
+    public void start() {
+        String resultIndex = anomalyDetector.getResultIndex();
+        try {
+            // use default detector result index which is system index
+            if (resultIndex == null) {
+                createOrUpdateDetector();
+                return;
+            }
+
+            // user custom result index
+            anomalyDetectionIndices.initCustomResultIndexAndExecute(resultIndex, () -> createOrUpdateDetector(), listener);
+        } catch (Exception e) {
+            logger.error("Failed to create index " + resultIndex, e);
```

Comment:
Maybe tune this error message? Seems the failures could be either issues creating the custom result index, as well as any errors found in `createOrUpdateDetector()`